cmake/all.cmake

```diff
@@ -128,6 +128,7 @@ foreach(filename
     list_append_deduplicate
     list_append_unique
     list_insert_in_workspace_order
+    list_insert_options
```

Comment:
Would `list_append_unique_options` make more sense for the name? Since the values are always appended. The only difference to the `list_append_unique` function above is that is only considers the key part and not the value when checking for existence.